components/controls/__tests__/buttons-test.tsx

```diff
@@ -19,7 +19,45 @@
  */
 import { shallow } from 'enzyme';
 import * as React from 'react';
-import { ButtonIcon, ButtonIconProps } from '../buttons';
+import { click, mockEvent } from '../../../helpers/testUtils';
+import { Button, ButtonIcon, ButtonIconProps } from '../buttons';
+
+describe('Button', () => {
+  it('should render correctly', () => {
+    const onClick = jest.fn();
+    const preventDefault = jest.fn();
+    const stopPropagation = jest.fn();
+    const wrapper = shallowRender({ onClick });
+    expect(wrapper).toMatchSnapshot();
+    click(wrapper.find('button'), mockEvent({ preventDefault, stopPropagation }));
+    expect(onClick).toBeCalled();
+    expect(preventDefault).toBeCalled();
+    expect(stopPropagation).not.toBeCalled();
+  });
+
+  it('should not stop propagation, but prevent default of the click event', () => {
+    const preventDefault = jest.fn();
+    const stopPropagation = jest.fn();
+    const wrapper = shallowRender({ preventDefault: false, stopPropagation: true });
+    click(wrapper.find('button'), mockEvent({ preventDefault, stopPropagation }));
+    expect(preventDefault).not.toBeCalled();
+    expect(stopPropagation).toBeCalled();
+  });
+
+  it('should disable buttons with a class', () => {
+    const onClick = jest.fn();
+    const button = shallowRender({ disabled: true, onClick }).find('button');
+    expect(button.props().disabled).toBeUndefined();
+    expect(button.props().className).toContain('disabled');
+    expect(button.props()['aria-disabled']).toBe(true);
+    click(button);
+    expect(onClick).not.toBeCalled();
+  });
+
```

Comment:
I'm wondering if we could/should test events on a disabled button, to cover the case that triggered this change in the first place?

Reply:
It would be nice, I just tried, but wasn't able to make the test fail in jest in fact and it seems legit since the issue happens only in Chrome currently...
So I don't know how to write such test so that it make sense...

Reply:
What I tried:
```ts
  it('should still get mouseover and mouseleave events when disabled', () => {
    const onMouseOver = jest.fn();
    const onMouseLeave = jest.fn();

    const wrapper = shallow(
      <Button
        disabled={true}
        // @ts-ignore
        onFocus={jest.fn()}
        onMouseOver={onMouseOver}
        onMouseLeave={onMouseLeave}
      />
    );
    wrapper.find('button').simulate('mouseover');
    expect(onMouseOver).toBeCalled();
    wrapper.find('button').simulate('mouseleave');
    expect(onMouseLeave).toBeCalled();
  });
```

Reply:
Ooh, `simulate('event')` is just a shortcut for `.props('onEvent')()`.. so yeah... that's never failing.
You would need to actually simulate the event like (thanks mdn!):
```
function simulateClick() {
  var evt = new MouseEvent("click", {
    bubbles: true,
    cancelable: true,
    view: window
  });
  var cb = document.getElementById("checkbox"); //element to click on
  var canceled = !cb.dispatchEvent(evt);
  if(canceled) {
    // A handler called preventDefault
    alert("canceled");
  } else {
    // None of the handlers called preventDefault
    alert("not canceled");
  }
}
```

Reply:
Thanks! I'll try!
But does it even make sense to do that ? I mean, the bug is on chrome only no ?

Reply:
Actually it seems it's not a bug, and React 16.13 explicitly gets rid of events on disabled buttons: https://reactjs.org/blog/2020/02/26/react-v16.13.0.html#notable-bugfixes

Reply:
Okay so I gave it a new try and still no luck.
I tried generating real events with `dispatchEvent` like in your example and adding event listener to the dom node, but it's the same it always trigger even for disabled buttons.

I'm giving up on this for now so that I can move forward with my PR on SC side. We can still do it later, ping me if you have new ideas and want us to try together!